rel: 0.27.0 (#536)
honeycombio/terraform-provider-honeycombio: CHANGELOG.md

# 0.27.0 (Aug 23, 2024)

NOTES:

> **This release contains breaking and behavioural changes**

Reviewer: obviously both spellings are right in their own way, but not sure which is right in this context. I guess more of a general docs style guide question that provider docs would follow?
```suggestion
**This release contains breaking and behavioral changes**
```

Author: Ha, my Canadian/British english (and spellchecker) is shining through. We use US spellings in our docs so I'll go with that: thanks for the catch!

* `r/dataset`: will now attempt to *delete* datasets on destroy.
  However, they come with delete protection (`delete_protection`) enabled by default and it must be disabled for the destroy to succeed.

>   This behaviour matches `r/envionment` as introduced in v0.25.0

Reviewer: same q as above

>   This behaviour matches `r/envionment` as introduced in v0.25.0

Reviewer: ```suggestion
  This behavior matches `r/environment` as introduced in v0.25.0
```

* `r/dataset`: will now error on create if a Dataset with that name already exists in the Environment.

>   Previously, the dataset would be silently imported into state, but given deletion is now possible on destroy this behaviour has to changed to be more standard.

Reviewer: same q here too

>   Previously, the dataset would be silently imported into state, but given deletion is now possible on destroy this behaviour has to changed to be more standard.

Author: ```suggestion
  Previously, the dataset would be silently imported into state, but given deletion is now possible on destroy this behavior has to changed to be more standard.
```

  The end-user can choose to either adjust the name or import the existing Dataset.
* client: if the Datasets API returns an HTTP '200 OK' -- as opposed to '201 Created' -- on create the `Datasets.Create` method will now return `ErrDatasetExists`.

ENHANCEMENTS:

> * *New Datasource*: `honeycombio_dataset` (#535)

Reviewer: ```suggestion
* *New Data Source*: `honeycombio_dataset` (#535)
```
